### ISSUE TITLE
Add pronoun field alongside bio

### ISSUE THREAD

**Issue by clarcharr** [tootsuite/mastodon#3211]
It'd be really nice if we could have some sort of explicit pronoun field in profiles. It was suggested before alongside other bio feature requests but all the relevant issues appear to be closed. So, I'll open a dedicated issue for it.

What I was thinking:

* It would offer checkboxes for a few presets, plus a free text field limited to a small number of characters.
* The presets would likely be she/her/her, he/him/his, and they/them/their, and could be translated.
* It would be an optional field, and not checking anything is equivalent to "address by name only."
* Checking "other" would automatically count as checking "they" on languages that are not the user's language.
* Languages can interpret "they" as "address by name only" if they don't have a standard-accepted equivalent (for example, French).
* Languages can choose to interpret "he" and "she" as "they" if they don't have any gendered pronouns.
* If we choose to interpret not checking any pronouns as not showing anything, we should have a separate checkbox for "name only."

The language for this field would probably be something similar to "address by" rather than "pronouns."

If it's necessary to get users to buy into it, we could probably do what Facebook does and list it as "male/female/other" and then only show the checkboxes if someone selects other, picking he/she for male/female.

Basically what I had in mind was having checkboxes like:

* he
* she
* they
* other: __________

* * * *

- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by Cassolotl** [comment]
What would the pronoun field be for? Would it be for the site to use to correctly gender members, like "so-and-so updated their status"? Or would it be for members to correctly gender each other?

Regarding the "other" option... With pronouns **in English**, in order for computers and people to use the pronouns correctly you need to know all five forms - like, for he/him it's actually he/him/his/his/himself.

- **He** is a person
- This tea is for **him**
- This is **his** tea
- The tea is **his**
- He made tea for **himself**

In other languages there are different grammar structures, that work differently. Pronouns can have more or less than five forms, that are used in very different ways and interact with verbs differently. Some languages have no gendered pronouns at all, and some have no gender-neutral pronouns. Localisation could be... interesting, and very complex?

If a pronoun field on Mastodon is just so that members can correctly gender each other, and there is an "other" option, I would recommend having a form a bit like [this](https://docs.google.com/spreadsheets/d/1iN-PY-LZU3N1QRuoM__HlH954xbxvcRfyp3eeUB9shA/edit?usp=sharing) on the "edit profile" page. There would need to be five fields, each with example sentences so that users can know that they're entering the right pronoun in the right box.

![screen shot 2017-05-21 at 23 13 13](https://cloud.githubusercontent.com/assets/6607512/26287838/17d8dc90-3e7b-11e7-8dd2-d61e0843088c.png)

In English there would also have to be a place to choose whether the verbs should be singular or plural - compare "he is" (singular verbs) and "they are" (plural verbs), for example. Should we say "xey is a happy person" or "xey are a happy person"? (Reminder: This is not the place to discuss whether singular they and other pronoun sets that use plural verbs are "correct" when used for one person. People use them that way, end of.)

That would result in something like "Pronouns: he/him/his/his/himself (singular verbs)" on a profile, which is very logical and complete but also requires some mental gymnastics for a curious user to actually find useful. Most people would not intuitively know how to use "ne/nim/nir/nirs/nimself (singular verbs)" just by reading the five forms presented in this way. So perhaps:

Pronouns: ne/nim/nir/nirs/nimself (singular verbs) [click here to see example sentence]

Example sentence: This is Sam. Ne brought nir favourite book with nim to the book club. Ne put a label it nimself, so everyone knows it's nirs.

(Or something better than that - I'm bad at example sentences!)

~

The starfriends.org [pronoun system](http://strfrnds.tumblr.com/post/156950927612/how-to-starfriends-alpha-pronouns) is spectacular but also probably a lot more complex than is usually needed. Starfriends (the site) needs to use pronouns when talking about members sometimes, *and also* shows users pronouns in places like in the inbox and on tooltips, so that members can correctly gender each other easily.

**Comment by clarcharr** [comment]
Wow, that's a lot more analysis than I expected. I'm glad. :)

The simplicity of this is mostly because I would rather have *something* and usually when I suggest these things I'm not 100% sure what people are willing to implement. For the sake of English pronouns, I've actually been trying to come up with a [system](https://docs.charr.xyz/pronouns/pronouns/) myself for properly listing pronouns in a programmatic way.

The main purpose of this, I think, would be simply so that users can figure out how to gender others correctly, which is why I figured that "other" would be sufficient, especially if it's language-specific. I figure that someone who uses ne/nem/nir in English for example would be totally fine defaulting to name-only in languages like French. Similarly, if someone uses she/her/her in English, then I would assume that they'd be fine with elle/la in French.

I highly doubt that the text on Mastodon will ever be complex enough to require programmatic conjugation of pronouns on a per-user basis, but if that's a likely path, it'd be nice to develop a robust system.

**Comment by Cassolotl** [comment]
>The simplicity of this is mostly because I would rather have something and usually when I suggest these things I'm not 100% sure what people are willing to implement.

I know what you mean. :) I'm not a developer, I mostly hover here to see what new features will happen and report things that annoy me, so I was not really expecting anyone else to comment in the same way as me. I was excited to see two of my favourite things in combination - Mastodon and pronouns/nonbinary things! Most of the time when I ask for pronoun things on websites it is not enthusiastically received, which is sad, because it would make a huge difference to my quality of life. For example, I spend time on Ravelry.com's forums. It's a knitting site, so a LOT of the members are women. My avatar has my pronouns in it, because that was the only way I could find to display my pronouns very obviously. I asked them for a pronouns field even just on profiles, and there was a huge backlash - they don't ask for gender because it's irrelevant and personal, so pronouns (which are public and assumed) are apparently the same thing, said the user base. People are not ready yet, especially user bases that skew a little older than my age.

Anyway! On topic, plz, self.

>The main purpose of this, I think, would be simply so that users can figure out how to gender others correctly, which is why I figured that "other" would be sufficient, especially if it's language-specific.

Yeah, I think you're probably right. If it was just a blank text box, and people can write in whatever, they can say "EN: they // FR: il // JP: 彼女" or "switch between he and she" or "any is fine" or whatever. And if someone put something super vague like "ne/nem", everyone would be like o.O, and then politely avoid pronouns for that person because they can't work out how to use nir pronouns correctly, and that's okay! :P Maybe someone would say "can you add the other three parts of your pronouns to the bio? Also, ne is or ne are?"

We're very early in pronoun-stuff. Maybe a plain text field is all we need?

**Comment by Kimik0** [comment]
Wow, that's impressively thorough. Having a pronoun field seems like a good idea because it helps make adding pronouns to one's bio more normal. At the same time I doubt that people will want to go through a whole form for this, even those who are enthusiastic about proper pronoun usage. I think a simple text field would work best.

**Comment by Cassolotl** [comment]
>At the same time I doubt that people will want to go through a whole form for this, even those who are enthusiastic about proper pronoun usage.

I would really like to find out if this is true!

Anyway, there are good ways of handling this. Starfriends.org, my favourite example, offers you a list based on what other people have entered, with the most common at the top, so 95% of people don't have to fill in a form:

![pronouns1](https://cloud.githubusercontent.com/assets/6607512/26304013/aadcb690-3ee1-11e7-8686-e1a4f4747935.jpg)

And if you have a new pronoun to add you can go on to the complex form:

![pronouns2](https://cloud.githubusercontent.com/assets/6607512/26304021/b529acf2-3ee1-11e7-87f5-bcd00607f944.png)

You can choose more than one pronoun, you can choose no pronouns, etc. If you're going to do pronouns properly you need to accommodate people with uncommon preferences, like (among others) "use this one and that one interchangeably" or "ask me at the start of our interaction each time, because it varies".

**Comment by averageunicorn** [comment]
I'm in favour of a pronoun field, and I would prefer a plain text field over an elaborate version for localisation reasons.

In German, my pronouns are basically "pick any of these 4 (also conjugate them the way you want)", and I know people whos pronoun choice is "any neopronoun is fine, and if that's hard for you, alternate between she and he". Those are hard to input with one simple form, so as long as pronoun fields are only there for users to look at and know which to use, a text box seems most convenient to me. The character limit shouldn't be too low, as some users will want to specify for 3 or more languages.

Additionally, while I'm glad that Mastodon only refers to users by pronoun in one place that I can think of (on profile pages: "You can follow them or interact with them if you have an account anywhere in the fediverse"), should that ever change, I'd be a bit more in favour of more complicated pronoun systems, simply because nobody should have to rely on translators to pick a comfortable default. I don't know if the "you can follow them…" on profiles is part of normal localisation or instance specific, but I've seen it translated both as "him/her" (French) and as "him" (German), and having that, especially the latter, all over Mastodon wouldn't be fun.

**Comment by navihawk** [comment]
If we're going to implement something like this, there should be the option either at a user level or shard level to enable or disable this feature, as there are communities on some shards that either:
A) don't care about a feature like this
B) oppose having to remember hundreds of superfluous pronouns
C) Want this feature

Besides, all of this can be listed in the bio section when you're typing it out. just saying.

**Comment by navihawk** [comment]
Another point that people I know have brought up is that trying to remember an additional 30+ pronouns, plus how many additional invented pronouns need to be memorized is very inefficient, and wastes a user's time whenever they have to go back to check someone's "pronouns" field just because they're afraid to offend them.

**Comment by Cassolotl** [comment]
If you want to talk to/about someone respectfully, you have to use the right pronouns. Nonbinary people are coming out and being out and existing more obviously in the Western world now, so using their pronouns correctly is just going to have to be a thing people do as nonbinary people become more out and accepted and celebrated. I guess if you don't like that you need to not follow or talk about nonbinary people? :P 

But yeah, dismissing neopronouns as inefficient, that's pretty disrespectful. We're humans with feelings, here.

**Comment by Cassolotl** [comment]
I really hope this thread doesn't become an argument about whether people should have pronouns other than he or she. This feature is requested because *people have and use pronouns other than he or she* and because people feel dysphoric when misgendered, cis people *and* trans people.

This isn't a discussion about whether it's necessary to list pronouns, it's about whether we should list them in our bios or have the option of a new field just for pronouns - so that it's easy to find them when you go looking.

**Comment by nightpool** [comment]
I've deleted two comments in this thread for being offtopic. To quote @Cassolotl:

> This isn't a discussion about whether it's necessary to list pronouns, it's about whether we should list them in our bios or have the option of a new field just for pronouns

**Comment by Gargron** [comment]
Mastodon UI does not use gendered pronouns to refer to people, and in situations when it would have to, we can always use the username instead of a pronoun. So it doesn't have to be a machine-readable field and it doesn't need all of these complicated conjugation tables, I can say as much.

**Comment by Cassolotl** [comment]
>in situations when it would have to, we can always use the username instead of a pronoun.

Doesn't it use masculine pronouns on userpages in German?

**Comment by navihawk** [comment]
@Cassolotl 

> admin setting: "pronoun field on user profiles? y/n"
This please.

> people feel dysphoric when misgendered, 

probably banned from ever commenting on tootsuite/mastodon for this, but [screw your fee-fees](https://youtu.be/e8chEy9rwL4?t=2m5s)
